src/App.tsx

```diff
@@ -24,72 +24,145 @@ enum SortType {
 type ReorderOptions = {
   sortType: SortType,
   isReversed: boolean,
+  lightAlpha: boolean,
+  lightLenght: boolean,
+  lightreset: boolean,
 };
 
-// Use this function in the render method to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
+  visibleGoods.sort((good1: string, good2: string) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+export class App extends React.Component <{}, ReorderOptions> {
+  state:ReorderOptions = {
+    sortType: SortType.NONE,
+    isReversed: false,
+    lightAlpha: false,
+    lightLenght: false,
+    lightreset: false,
+  };
+
+  sortAlpha = () => {
+    this.setState({
+      sortType: SortType.ALPHABET,
+      lightAlpha: true,
+      lightLenght: false,
+    });
+  };
+
+  sortLenght = () => {
+    this.setState({
+      sortType: SortType.LENGTH,
+      lightLenght: true,
+      lightAlpha: false,
+    });
+  };
+
+  sortRevers = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+      lightreset: !state.isReversed,
+    }));
+  };
+
+  reset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+      lightAlpha: false,
+      lightLenght: false,
+      lightreset: false,
+    });
+  };
+
+  render() {
+    const visibleGoods = getReorderedGoods(goodsFromServer, this.state);
+
+    const showResetButton
+      = this.state.sortType !== SortType.NONE
+      || this.state.isReversed !== false;
```

Comment:
```suggestion
    const showResetButton = this.state.sortType !== SortType.NONE
      || this.state.isReversed !== false;
```
It would be great to find a better name for boolean variable. Now it may be confused with function
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig